### PR TITLE
FIXUP: rename Nilearn.masking.compute_gray_matter_mask

### PR DESCRIPTION
**Pull request by alpinho**
solve issue #2280

### CODE REVIEW

File: nilearn/masking.py [hunk 1]
```diff
@@ -519,6 +520,9 @@ def compute_multi_background_mask(data_imgs, border_size=2, upper_cutoff=0.85,
     return mask
 
 
+@deprecated("Function 'compute_gray_matter_mask' has been renamed to "
+            "'compute_whole_brain_matter_mask' and "
+            "'compute_gray_matter_mask' will be removed in future releases.")
```
Comment (bthirion) [inline nilearn/masking.py:525]: I guess in release 0.9

File: doc/whats_new.rst [hunk 2]
```diff
@@ -21,6 +21,10 @@ Fixes
 -----
 
 - :class:`nilearn.input_data.NiftiLabelsMasker` no longer ignores its `mask_img`
+- :class:`nilearn.masking.compute_gray_matter_mask` has been renamed to
```
Comment (bthirion) [inline doc/whats_new.rst:24]: this is a function

File: nilearn/masking.py [hunk 3]
```diff
@@ -562,6 +566,52 @@ def compute_gray_matter_mask(target_img, threshold=.5,
...
+
+
+def compute_brain_mask(target_img, threshold=.5, connected=True,
+                       opening=2, memory=None, verbose=0):
+    """Compute the whole-brain mask. This mask is calculated through the
+    resampling of the MNI152 template mask onto the target image.
+
+    Parameters
+    ----------
+    target_img: Niimg-like object
```
Comment (bthirion) [inline nilearn/masking.py:581]: We could ttake this opportunity to update docstrings, following #2450.

**Comment by bthirion** [comment]
I think that there are many other occurrence of `compute_gray_matter_mask` throughout the library that should be fixed.

File: doc/whats_new.rst (hunk 2 above)
Comment (GaelVaroquaux) [inline doc/whats_new.rst:24]:
```suggestion
- :func:`nilearn.masking.compute_gray_matter_mask` has been renamed to
```

File: nilearn/masking.py [hunk 4]
```diff
@@ -562,6 +566,52 @@ def compute_gray_matter_mask(target_img, threshold=.5,
...
+        See http://nilearn.github.io/manipulating_images/input_output.html
+        Images used to compute the mask. 3D and 4D images are accepted.
+        Only the shape and affine of target_img will be used here.
+
+    threshold: float, optional
```
Comment (GaelVaroquaux) [inline nilearn/masking.py:586]:
```suggestion
    threshold : float, optional
```
The doc standards actually ask for a space between the attribute name and the colon (we did it wrong in the past)

**Comment by jeromedockes** [comment]
also don't forget to update nilearn/tests/test_masking.py :)

File: doc/whats_new.rst (hunk 2 above)
Comment (alpinho) [reply doc/whats_new.rst:24]: Thanks for the comments. I am solving them. Should I change also the name of the function `test_compute_gray_matter_mask`?: https://github.com/nilearn/nilearn/blob/02b6ae5b0a45a4ac3a8644360a652bb0515b337d/nilearn/input_data/tests/test_nifti_masker.py#L317

**Comment by alpinho** [comment]
Thanks for the comments. I am trying to address all of them. Should I change also the name of the function `test_compute_gray_matter_mask`?: https://github.com/nilearn/nilearn/blob/02b6ae5b0a45a4ac3a8644360a652bb0515b337d/nilearn/input_data/tests/test_nifti_masker.py#L317

**Comment by emdupre** [comment]
>  Should I change also the name of the function `test_compute_gray_matter_mask`?: 

Yes, please !

**Comment by jeromedockes** [comment]
yes and have one that looks like `test_deprecation_warning_compute_gray_matter_mask`.

**Comment by thomasbazeille** [comment]
The 2 Travis failures seem unrelated : one is neurosynth being down apparently and the other one nitrc.

```Downloading data from https://www.nitrc.org/frs/download.php/9343/lh.aparc.a2009s.annot ...
FAILED
plotting/tests/test_html_surface.py::test_view_img_on_surf <- ../home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/nilearn/plotting/tests/test_html_surface.py opened file:///tmp/nilearn_plot_d1d2cdtl.html
removed /tmp/nilearn_plot_d1d2cdtl.html
opened file:///tmp/nilearn_plot_mu41foaa.html
removed /tmp/nilearn_plot_mu41foaa.html
opened file:///tmp/nilearn_plot_2on1o82_.html
removed /tmp/nilearn_plot_2on1o82_.html
opened file:///tmp/nilearn_plot_cq8ob1xf.html
removed /tmp/nilearn_plot_cq8ob1xf.html
Dataset created in /tmp/pytest-of-travis/pytest-0/test_get_dataset_dir2/nilearn_shared_data/fsaverage
Downloading data from https://www.nitrc.org/frs/download.php/10846/fsaverage.tar.gz ...
FAILED```


```
Downloading file: http://neurosynth.org/api/decode/?neurovault=a
Problem downloading file from http://neurosynth.org/api/decode/?neurovault=a
Could not fetch words for image a
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/nilearn/datasets/neurovault.py", line 1576, in _download_image_terms
    verbose=download_params['verbose'])
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/nilearn/datasets/neurovault.py", line 1186, in _simple_download
    overwrite=True, verbose=0)
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/nilearn/datasets/tests/test_neurovault.py", line 375, in _fail
    raise neurovault.URLError('problem')
urllib.error.URLError: <urlopen error problem>```

**Comment by thomasbazeille** [comment]
otherwise it seems ready for a new round of review !

File: nilearn/input_data/tests/test_nifti_masker.py [hunk 5]
```diff
@@ -314,9 +315,18 @@ def test_compute_epi_mask():
                              get_data(mask4)[3:12, 3:12])
 
 
+@deprecated("Function 'test_compute_gray_matter_mask' has been renamed to "
+            "'test_compute_brain_mask' and "
+            "'test_compute_gray_matter_mask' will be removed in release 0.9.")
 def test_compute_gray_matter_mask():
     # Check masker for template masking strategy
 
+    return test_compute_brain_mask()
```
Comment (GaelVaroquaux) [inline nilearn/input_data/tests/test_nifti_masker.py:324]: I don't think that we need a deprecated test. This function can simply be removed.

**Comment by GaelVaroquaux** [comment]
Also, CircleCI has a few valid errors:
<pre>
sphinx.errors.SphinxWarning: /home/circleci/project/doc/whats_new.rst:24:py:func reference target not found: nilearn.masking.compute_gray_matter_mask
</pre>
I would suggest to do a `git grep compute_gray_matter_mask` in the docs folder, to check that there a not longer occurences of compute_gray_matter_mask

**Comment by alpinho** [comment]
> Also, CircleCI has a few valid errors:
> 
> sphinx.errors.SphinxWarning: /home/circleci/project/doc/whats_new.rst:24:py:func reference target not found: nilearn.masking.compute_gray_matter_mask
> 
> I would suggest to do a `git grep compute_gray_matter_mask` in the docs folder, to check that there a not longer occurences of compute_gray_matter_mask

I have addressed this problem by changing the description of the new function in the [`whats_new` doc (lines 24-26)](https://github.com/nilearn/nilearn/blob/327179a486da139e618cce9d37b97f37487817cf/doc/whats_new.rst).

**Comment by jeromedockes** [comment]
ThomasBaz <notifications@github.com> writes:

> ```
> Downloading file: http://neurosynth.org/api/decode/?neurovault=a
> Problem downloading file from http://neurosynth.org/api/decode/?neurovault=a
> Could not fetch words for image a
> Traceback (most recent call last):
>   File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/nilearn/datasets/neurovault.py", line 1576, in _download_image_terms
>     verbose=download_params['verbose'])
>   File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/nilearn/datasets/neurovault.py", line 1186, in _simple_download
>     overwrite=True, verbose=0)
>   File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/nilearn/datasets/tests/test_neurovault.py", line 375, in _fail
>     raise neurovault.URLError('problem')
> urllib.error.URLError: <urlopen error problem>```
>

This is logged by the fetcher but does not result in a failure

File: nilearn/input_data/tests/test_nifti_masker.py (hunk 5 above)
Comment (alpinho) [reply nilearn/input_data/tests/test_nifti_masker.py]: I have removed the deprecated test but not the `test_compute_brain_mask()`, if that's what you meant.

**Comment by GaelVaroquaux** [comment]
> I have removed the deprecated test but not the test_compute_brain_mask(), if that's what you meant.

Yes

**Comment by jeromedockes** [comment]
hey @alpinho I opened a PR on your branch: https://github.com/alpinho/nilearn/pull/1

**Comment by alpinho** [comment]
Thanks @jeromedockes . I just committed the fixes you have addressed.

File: nilearn/masking.py [hunk 6]
```diff
@@ -521,7 +521,7 @@ def compute_multi_background_mask(data_imgs, border_size=2, upper_cutoff=0.85,
 
 
 @deprecated("Function 'compute_gray_matter_mask' has been renamed to "
-            "'compute_whole_brain_matter_mask' and "
+            "'compute_brain_matter_mask' and "
```
Comment (jeromedockes) [inline nilearn/masking.py:524]:
```suggestion
            "'compute_brain_mask' and "
```

**Comment by jeromedockes** [comment]
looks good! will merge as soon as circleci finishes

**Comment by jeromedockes** [comment]
thanks @alpinho !! merging

**Comment by alpinho** [comment]
Thanks @jeromedockes!!! @bthirion @GaelVaroquaux If you all agree, perhaps a new issue about creating specific masks (i.e. gray-matter mask or white-matter masks) should be created. One option is to allow this renamed function to have new parameters that will perform these specific estimations.

**Pull request merged**